### ISSUE TITLE
Phase out 0.10, 0.12 support

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#230]
Node 0.10 is in maintenance mode. 
Node 0.12 enters maintenance mode on 4/1. 

Given that nvm exists, it is not clear to me that the polyfills and Babel steps required to support 0.x are necessary. Convenient, maybe, but not strictly necessary. 

Would love feedback on this. Anyone expect to be significantly impacted if I drop 0.x Node support?

**Comment by le0nik** [comment]
I think it's better to align dropping 0.x support with `esint`. See: https://github.com/eslint/eslint/issues/4483.

Also i tried to use `memo-parser` in my config and saw my build fail on Node 0.10 and 0.12. Is it possible to rewrite it in ES5? Will you accept a pull-request?

**Comment by benmosher** [comment]
I'm on the fence.

It's nearly April. I can appreciate the practical aspects of keeping the compatibility, but what about doing something like:
```bash
nvm exec 4 eslint ...
```
?

Or even only linting on the Node 4+ builds? something like
```bash
[[ `node --version` > 'v4.'* ]] && eslint ...
```

It wouldn't be tough to modify the memo-parser to be 0.10-compatible, sure. I'm just not sure it is worthwhile, long term, given the state of LTS and the fact that no one is running ESLint "in production". And now seems like as good a time as any.

I'm excited to start stripping away Babel transformers and get down to more native code. I think I could almost get away with just the loose modules transform on Node v5.9.

**Comment by jfmengels** [comment]
I agree with @le0nik, this should align with what `eslint` supports, otherwise it will make configuring difficult for users. If it were up to me, I'd probably change the major version when `eslint` releases v3, and say that the next version needs node v4, and previous major versions support 0.x.

I understand your frustration as I have the same ones (though I usually don't transpile with Babel in my modules, so I'm still using plain ES2015).

(FYI, we still use 0.12 at work, though we want to hop to v6 soon).

**Comment by benmosher** [comment]
mm, fair enough. I'll have to revert the things I ripped out for `eslint-module-utils` but that's not the end of the world.

**Comment by jfmengels** [comment]
Would be nice if ESX features were somehow not restricted by the Node version... We'll have the same concerns when v4 goes out of maintenance, v6 then, etc... (see how long we'll have to wait before `async/await` will be out of maintenance :p)

**Comment by benmosher** [comment]
done in v2 branch.
